.github/workflows/deploy_devnet.yml

```diff
@@ -0,0 +1,230 @@
+              exit 1
+            else
+              sleep 60  # Wait for 30 seconds before checking again
+            fi
+          done
+
+      - name: Configure devnet
+        id: configure_devnet
+        run: |
+          COMMAND_ID=$(aws ssm send-command \
+            --instance-ids ${{ secrets.DEVNET_INSTANCE_ID }} \
+            --document-name "AWS-RunShellScript" \
+            --parameters '{"commands":[
+                "sudo cp -rp /root/.cargo/bin/miden-node /usr/bin/miden-node && sudo chmod a+x /usr/bin/miden-node",
+                "sudo cp -rp /root/.cargo/bin/miden-faucet /usr/bin/miden-faucet && sudo chmod a+x /usr/bin/miden-faucet",
+                "sudo /usr/bin/miden-node init -c /etc/miden/miden-node.toml -g /opt/miden/miden-node/genesis.toml",                
+                "sudo /usr/bin/miden-node make-genesis -i /opt/miden/miden-node/genesis.toml -o /opt/miden/miden-node/genesis.dat",
+                "sudo chown -R miden:miden /opt/miden/miden-node",
+                "sudo /usr/bin/miden-faucet init -c /opt/miden/miden-faucet/miden-faucet.toml",
+                "sudo chown -R miden:miden /opt/miden/miden-faucet"
+              ]}' \
+            --output text \
+            --query "Command.CommandId")
+          echo "command_id=$COMMAND_ID" >> $GITHUB_OUTPUT
+
+      - name: Check configure devnet command status and retrieve output
+        run: |
+          sleep 20
+          STATUS=$(aws ssm list-command-invocations \
+              --command-id ${{ steps.configure_devnet.outputs.command_id }} \
+              --details \
+              --query "CommandInvocations[0].Status" \
+              --output text)
+          echo "Command Status: $STATUS"
+          OUTPUT=$(aws ssm list-command-invocations \
+              --command-id ${{ steps.configure_devnet.outputs.command_id }} \
+              --details \
+              --query "CommandInvocations[0].CommandPlugins[0].Output" \
+              --output text)
+          echo "Command Output: $OUTPUT"
+          if [ "$STATUS" != "Success" ]; then
+              echo "Command failed with status: $STATUS"
+              exit 1
+          fi
+
+      - name: Start services
+        id: start_services
+        run: |
+          COMMAND_ID=$(aws ssm send-command \
+            --instance-ids ${{ secrets.DEVNET_INSTANCE_ID }} \
+            --document-name "AWS-RunShellScript" \
+            --parameters '{"commands":[
+                "sudo systemctl daemon-reload",
+                "sudo systemctl enable miden-node",
+                "sudo systemctl enable miden-faucet",
+                "sudo systemctl start miden-node",
+                "sleep 5",
+                "sudo systemctl start miden-faucet"
+              ]}' \
+            --output text \
+            --query "Command.CommandId")
+          echo "command_id=$COMMAND_ID" >> $GITHUB_OUTPUT
+
+      - name: Check Start services command status and retrieve output
```

Comment:
Is it possible to somehow extract these checks steps? Probably not from what I remember without creating a custom github action?

Reply:
Looks like it should be possible to create a nice wrapper using a composite action. Though I can attempt that at some other time. Right now its more important to just get this up and running.